### PR TITLE
12 app metadata

### PR DESCRIPTION
**Pull request by rkzel**
Solved issues:

#17 - getAppMetadata handler add wrapper along with appropriate test
#14 - Create appMetadata observable and init function
#15 - Create registerAppMetadata handler in wrapper
#16 - create getAppMetadata endpoint in API

### CODE REVIEW

File: packages/aragon-wrapper/src/index.js [hunk 1]
```diff
@@ -859,6 +859,32 @@ export default class Aragon {
...
+   * registers new app metadata item
+   *
+   * @param {string} from
+   * @param {string} dataId
+   * @param {string} cid
+   */
+  registerAppMetadata (from, dataId, cid) {
+    this.registerMetadata.next({
+      from,
+      to: ['*'],
```
Comment (topocount) [inline packages/aragon-wrapper/src/index.js]: This looks like it's hard-coded, and can't be change by passing in a custom "to" param

File: packages/aragon-wrapper/src/index.test.js [hunk 2]
```diff
@@ -1468,3 +1468,17 @@ test.only('should set forwarded actions', async (t) => {
     }])
   })
 })
+
+test('should init the appMetadata correctly', async (t) => {
```
Comment (topocount) [inline packages/aragon-wrapper/src/index.test.js:1547]: Please add a test here to ensure a "to" param can be passed and read from at this layer, and the wrapper implementation behaves as expected in other ways. There should be more than just an initialization test. Make sure edge cases behave as expected, etc.

File: packages/aragon-api/src/index.js [hunk 3]
```diff
@@ -391,18 +391,27 @@ export class AppProxy {
...
    *
    * @param {string} from Address of the application generating the data
-   * @param {<Array>string} to Address of the applications allowed access to the data
+   * @param {<Array>string} to Optional list of addresses of the applications allowed access to the data, defaults to '*'
    * @param {string} dataId internal ID assigned to the data by the originator
    * @param {string} cid external identifier (e.g., IPFS hash)
    * @return {void}
    */
-  registerAppMetadata (from, to, dataId, cid) {
+  registerAppMetadata (from, to = ['*'], dataId, cid) {
```
Comment (topocount) [inline packages/aragon-api/src/index.js]: This default `'*'` should be tested in the api layer as well as the wrapper layer. to ensure publicly broadcast metadata is accessible to all queries

**Review by topocount — CHANGES_REQUESTED** [review]
This is a good start, but it looks like the "to" parameter is hard-coded at the wrapper layer, and the wrapper layer needs more tests as well.

File: packages/aragon-wrapper/src/index.js (hunk 1 above)
Comment (rkzel) [reply packages/aragon-wrapper/src/index.js]: Changed from hardcoded to default parameters in both wrapper and api. Not sure if it's not redundant..

File: packages/aragon-wrapper/src/index.test.js (hunk 2 above)
Comment (rkzel) [reply packages/aragon-wrapper/src/index.test.js:1547]: Done.

File: packages/aragon-api/src/index.js (hunk 3 above)
Comment (rkzel) [reply packages/aragon-api/src/index.js]: Done.

**Review by topocount — APPROVED** [review]
